### PR TITLE
COMMUNITY-ROLES: fix note, remove broken link reference

### PR DESCRIPTION
**Pull request by kbdharun**
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

## Changes

- Fix the broken note rendering.
- Remove the merged PR check URL (`https://github.com/tldr-pages/tldr/commits?committer=<username>`) as afaik GitHub no longer offers a direct way to check it. In future, we can continue manually checking the merged PRs (for this requirement).

> [!NOTE]
> As the wording signifies it is for checking merged commits by you, not to be confused with authored/coauthored commits `https://github.com/tldr-pages/tldr/commits?author=<username>`

### CODE REVIEW

**Review by acuteenvy — APPROVED** [review]

File: COMMUNITY-ROLES.md [hunk 1]
```diff
@@ -35,7 +35,6 @@ exceptions can always be considered through open community discussion.)
   (Maintenance work means facilitating contributions by other people,
   which in this project typically consists of reviewing and/or merging PRs.)
   Specifically: once a repository collaborator has _merged at least 10 PRs_
-  (see `https://github.com/tldr-pages/tldr/commits?committer=<username>`)
```
[the remark this answers is not in the document]
Comment (kbdharun) [reply COMMUNITY-ROLES.md:38]: I was expecting this question xD, that's why I mentioned it in the PR description https://github.com/tldr-pages/tldr/pull/11915#issue-2060031202.

**TLDR.** It says authored commits not commits you merged (i.e. by others including your authored commits)

[the remark this answers is not in the document]
Comment (kbdharun) [reply COMMUNITY-ROLES.md:38]: > https://github.com/tldr-pages/tldr/commits/main/?author=

This is the same, it doesn't tell the list of commits I was a committer in.
